x-pack/plugins/apm/public/context/breadcrumbs/use_breadcrumb.ts

```diff
@@ -38,5 +38,5 @@ export function useBreadcrumb(breadcrumb: Breadcrumb | Breadcrumb[]) {
       }
     };
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [match]);
```

Comment:
no, the problem keeps happening with `match?.route`

Reply:
I think the match object itself is recreated every render. Maybe we shouldn't use useEffect at all 🤔

Reply:
I made some tests, and it looks like the `useEffect` is not needed indeed.

Reply:
@cauemarcondes in this case you can also remove the array entirely, I think. But in all honesty we should probably understand how we should use this hook. Perhaps we need a similar approach to `useFetcher` where we are able to define the dependencies.

Reply:
`useEffect(..., [ match ])` is the same as `useEffect(...)`, because `match` is recreated every render, thus invalidating the dependencies of `useEffect` every render. Just to clarify, did you use `useEffect` without dependencies, or did you remove the hook entirely?

Reply:
One thing though, we can't guarantee that the breadcrumbs only depend on route parameters. E.g., there might be a dependency on a data call. So maybe we should explicitly define the dependencies in the hook.

Reply:
I added an optional parameter so consumers can specify their own dependencies which is going to invalidate the useEffect. https://github.com/elastic/kibana/pull/136144/commits/411c81b489b2d62a4a212651ce44c4fdea944abd. Do you know anywhere where I must use it on our code?

Reply:
Why is it optional? You'd also need to add it to `.eslintrc.js`: https://github.com/elastic/kibana/blob/33b743c84e9d27e6a0234ebea85964ce0ddefb24/.eslintrc.js#L982

Reply:
because in most cases we don't need to pass any other dependency to this hook, right?!

Reply:
Only if they're entirely static. But in that case they should pass an empty array. I'd rather be explicit about that.